client/dashboard/sites/overview/site-card.tsx

```diff
@@ -102,40 +103,72 @@ function FieldTitle( { children }: { children: React.ReactNode } ) {
 	);
 }
 
-function PlanDetails( { site, currentPlan }: { site: Site; currentPlan: Plan } ) {
-	if ( ! site.plan || ! currentPlan ) {
+function PlanDetails( { site }: { site: Site } ) {
+	const { data: currentPlan } = useQuery( siteCurrentPlanQuery( site.slug ) );
+
+	if ( ! site.plan ) {
 		return null;
 	}
 
 	const {
 		plan: { product_name_short, is_free: isFree },
 	} = site;
-	const { expiry, id } = currentPlan;
+
 	return (
 		<VStack>
 			<FieldTitle>{ __( 'Plan' ) }</FieldTitle>
 			{ product_name_short && <Text>{ product_name_short }</Text> }
-			<Text>{ getPlanExpirationMessage( { isFree, expiry } ) }</Text>
-			{ id ? (
-				<Button href={ `/purchases/subscriptions/${ site.slug }/${ id }` } variant="link">
-					{ __( 'Manage subscription' ) }
-				</Button>
-			) : (
+			<Text>
+				<PlanExpirationMessage isFree={ isFree } currentPlan={ currentPlan } />
```

Comment:
(curious) why do you prefer React component here rather than the original regular function? I see that you like to do that in many places.

I usually just write something like `renderForm()`, `renderNotice()`. Is this just preference or is there a good reason to prefer one than the other